AOT Webpack build fail: Properties are private

This repository's issues are reserved for feature requests and bug reports.

### Do you want to request a *feature* or report a *bug*?

Bug

#### Bug Report

When trying to execute AOT compilation using webpack a number of files won't compile as properties are marked as private: Errors such as the following are received:

> Error in bail mode: [at-loader] compiled/node_modules/@covalent/core/data-table/data-table.component.ngfactory.ts:200:48 
>     TS2341: Property '_templates' is private and only accessible within class 'TdDataTableComponent'.

This happens on the `@ViewChild()` and `@ContentChildren()` properties so to fix the problem, the following properties need to be marked as public: 

`_templates` in core/file/file-input/data-table.component.ts. (line 72)
`_inputElement` in in core/file/file-input/file-input.component.ts (line 52)
`_drawerMenu` in core/layout/navigation-drawer/navigation-drawer.component.ts (line 35)
`_searchInput` in core/search/search-box/search-box.component (line 28)
`_input` in core/search/search-input/search-input.component (line 28)
`_content` in core/steps/step.component (line 54)

To build my project, I am using the Webpack scripts found here:

[https://github.com/AngularClass/angular2-webpack-starter](https://github.com/AngularClass/angular2-webpack-starter)

##### Screenshots or link to CodePen/Plunker/JSfiddle

N/A

#### What is the expected behavior?

N/A

#### What is the motivation / use case for changing the behavior?

N/A

#### Which version of Angular and Material, and which browser and OS does this issue affect?

Versions from package.json: 

angular: 2.4.7
covalent/core: ^1.0.0-beta.2

##### Other information
(e.g. detailed explanation, stacktraces, related issues, suggestions how to fix)

have not tested the optional or external component libraries.

@emoralesb05 how do I get that fix from develop?
1. I cloned the project 
2. Run `npm run build` to generate the platform packages. 
3. But when I do a local install I get errors like these:

```
ERROR in Error encountered resolving symbol values statically. Could not resolve @angular/animations relative to /Users/pdejuan/projects/sampleapp/frontend/node_modules/@covalent/core/common/animations/collapse/collapse.animation.d.ts., resolving symbol TdCollapseAnimation in /Users/pdejuan/projects/sampleapp/frontend/node_modules/@covalent/core/common/animations/collapse/collapse.animation.d.ts, resolving symbol TdCollapseAnimation in /Users/pdejuan/projects/sampleapp/frontend/node_modules/@covalent/core/common/animations/collapse/collapse.animation.d.ts, resolving symbol TdExpansionPanelComponent in /Users/pdejuan/projects/sampleapp/frontend/node_modules/@covalent/core/expansion-panel/expansion-panel.component.d.ts, resolving symbol TdExpansionPanelComponent in /Users/pdejuan/projects/sampleapp/frontend/node_modules/@covalent/core/expansion-panel/expansion-panel.component.d.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/Users/pdejuan/projects/sampleapp/frontend/src'
```

--- a/src/platform/core/data-table/data-table.component.ts
+++ b/src/platform/core/data-table/data-table.component.ts
@@ -69,7 +69,7 @@ export class TdDataTableComponent implements ControlValueAccessor, AfterContentI
 
   /** template fetching support */
   private _templateMap: Map<string, TemplateRef<any>> = new Map<string, TemplateRef<any>>();
-  @ContentChildren(TdDataTableTemplateDirective) private _templates: QueryList<TdDataTableTemplateDirective>;
+  @ContentChildren(TdDataTableTemplateDirective) _templates: QueryList<TdDataTableTemplateDirective>;
 
   /**
    * Implemented as part of ControlValueAccessor.

--- a/src/platform/core/file/file-input/file-input.component.ts
+++ b/src/platform/core/file/file-input/file-input.component.ts
@@ -49,7 +49,7 @@ export class TdFileInputComponent implements ControlValueAccessor {
   private _disabled: boolean = false;
 
   /** The native `<input type="file"> element */
-  @ViewChild('fileInput') private _inputElement: ElementRef;
+  @ViewChild('fileInput') _inputElement: ElementRef;
   get inputElement(): HTMLInputElement {
     return this._inputElement.nativeElement;
   }

--- a/src/platform/core/layout/navigation-drawer/navigation-drawer.component.ts
+++ b/src/platform/core/layout/navigation-drawer/navigation-drawer.component.ts
@@ -32,7 +32,7 @@ export class TdNavigationDrawerComponent implements OnInit, OnDestroy {
     return this._menuToggled;
   }
 
-  @ContentChildren(TdNavigationDrawerMenuDirective) private _drawerMenu: QueryList<TdNavigationDrawerMenuDirective>;
+  @ContentChildren(TdNavigationDrawerMenuDirective) _drawerMenu: QueryList<TdNavigationDrawerMenuDirective>;
 
   /**
    * Checks if there is a [TdNavigationDrawerMenuDirective] as content.

--- a/src/platform/core/search/search-box/search-box.component.ts
+++ b/src/platform/core/search/search-box/search-box.component.ts
@@ -27,7 +27,7 @@ import { TdSearchInputComponent } from '../search-input/search-input.component';
 export class TdSearchBoxComponent {
 
   private _searchVisible: boolean = false;
-  @ViewChild(TdSearchInputComponent) private _searchInput: TdSearchInputComponent;
+  @ViewChild(TdSearchInputComponent) _searchInput: TdSearchInputComponent;
 
   set value(value: any) {
     this._searchInput.value = value;

--- a/src/platform/core/search/search-input/search-input.component.ts
+++ b/src/platform/core/search/search-input/search-input.component.ts
@@ -29,7 +29,7 @@ import 'rxjs/add/operator/debounceTime';
 })
 export class TdSearchInputComponent implements OnInit {
 
-  @ViewChild(MdInputDirective) private _input: MdInputDirective;
+  @ViewChild(MdInputDirective) _input: MdInputDirective;
 
   value: string;
 

--- a/src/platform/core/steps/step.component.ts
+++ b/src/platform/core/steps/step.component.ts
@@ -51,7 +51,7 @@ export class TdStepComponent implements OnInit {
     return this._contentPortal;
   }
 
-  @ViewChild(TemplateRef) private _content: TemplateRef<any>;
+  @ViewChild(TemplateRef) _content: TemplateRef<any>;
   @ContentChild(TdStepLabelDirective) stepLabel: TdStepLabelDirective;
   @ContentChild(TdStepActionsDirective) stepActions: TdStepActionsDirective;
   @ContentChild(TdStepSummaryDirective) stepSummary: TdStepSummaryDirective;

